code/modules/power/tesla/energy_ball.dm

```diff
@@ -312,36 +313,46 @@
 			. = zapdir
 
 	//per type stuff:
+	var/range = 3
 	if(!QDELETED(closest_million_dollar_baby))
-		closest_million_dollar_baby.zap_act(power, zap_flags, shocked_targets)
+		power = closest_million_dollar_baby.zap_act(power, zap_flags, shocked_targets)
 
 	else if(!QDELETED(closest_tesla_coil))
-		closest_tesla_coil.zap_act(power, zap_flags, shocked_targets)
+		power = closest_tesla_coil.zap_act(power, zap_flags, shocked_targets)
+		range = 5
 
 	else if(!QDELETED(closest_grounding_rod))
-		closest_grounding_rod.zap_act(power, zap_flags, shocked_targets)
+		power = closest_grounding_rod.zap_act(power, zap_flags, shocked_targets)
 
 	else if(!QDELETED(closest_rideable))
-		closest_rideable.zap_act(power, zap_flags, shocked_targets)
+		power = closest_rideable.zap_act(power, zap_flags, shocked_targets)
 
 	else if(!QDELETED(closest_mob))
 		closest_mob.set_shocked()
 		addtimer(CALLBACK(closest_mob, /mob/living/proc/reset_shocked), 10)
-		var/shock_damage = (zap_flags & ZAP_MOB_DAMAGE)? (min(round(power/600), 90) + rand(-5, 5)) : 0
+		var/shock_damage = (zap_flags & ZAP_MOB_DAMAGE) ? (min(round(power/600), 90) + rand(-5, 5)) : 0
 		closest_mob.electrocute_act(shock_damage, source, 1, SHOCK_TESLA | ((zap_flags & ZAP_MOB_STUN) ? NONE : SHOCK_NOSTUN))
 		if(issilicon(closest_mob))
 			var/mob/living/silicon/S = closest_mob
 			if((zap_flags & ZAP_MOB_STUN) && (zap_flags & ZAP_MOB_DAMAGE))
 				S.emp_act(EMP_LIGHT)
-			tesla_zap(S, 7, power / 1.5, zap_flags, shocked_targets) // metallic folks bounce it further
+			range = 7 // metallic folks bounce it further
 		else
-			tesla_zap(closest_mob, 5, power / 1.5, zap_flags, shocked_targets)
+			range = 5
+		power /= 1.5
 
 	else if(!QDELETED(closest_machine))
-		closest_machine.zap_act(power, zap_flags, shocked_targets)
+		power = closest_machine.zap_act(power, zap_flags, shocked_targets)
 
 	else if(!QDELETED(closest_blob))
-		closest_blob.zap_act(power, zap_flags, shocked_targets)
+		power = closest_blob.zap_act(power, zap_flags, shocked_targets)
 
 	else if(!QDELETED(closest_structure))
-		closest_structure.zap_act(power, zap_flags, shocked_targets)
+		power = closest_structure.zap_act(power, zap_flags, shocked_targets)
+
+
+	var/zap_count = 1
+	if(prob(20))
+		zap_count++
+	for(var/i in 1 to zap_count)
```

Comment:
```suggestion
	for(var/i in 1 to pick(1, 2; 20))
```

Reply:
What wizardry is this.

Reply:
Uh, i'm pretty sure the format is other way around ? As in this is prob(2) of 20 ?

Reply:
oh oops yeah